### DISCUSSION TITLE
Accessing XML fields parsed from JSON field

### DISCUSSION THREAD

**Discussion by RPiAwesomeness** [mockoon/mockoon#1485] [Ideas]
I have a use-case where a request is coming in with valid XML in a JSON field - can the the `xml-js` parsing be exposed via a request helper?

I had the idea based on the  [Response Configuration > XML Support docs](https://mockoon.com/docs/latest/response-configuration/xml-support/) which only works if the request's Content-Type is set to `application/xml` or similar.

For example, with the following request:

POST /endpoint
Content-Type: application/json

```json
{
  "id": "test",
  "xml": "<?xml version="1.0"?>\n<definitions xmlns=\"beans\" xmlns:bpmndi=\"bpmdi\">\n  <process id=\"key\" isExec=\"true\" name=\"name\">\n    <documentation>docs</documentation>\n    <dataObject id="version" name="version">\n      <extensionElements>\n        <flowable:value>VERSION</flowable:value>\n      </extensionElements>\n    </dataObject>\n  </process>\n  <bpmndi:BPMNDiagram>\n  </bpmndi:BPMNDiagram>\n</definitions>"
}
```

I would like to access a field in the XML via something like the following in my response template:

```json
{
  "id": "{{ body 'id' }}"
  "name": "{{ xml (body 'xml') 'definitions.process._attributes.name' }}"
}
```

**Comment by 255kb** [comment]
This is not possible yet. Though I'm not against creating a templating helper for this purpose. Feel free to open a new issue and submit a PR. Thanks!
